go/tool/bertydoctor/checkRDVP.go

```diff
@@ -0,0 +1,291 @@
+							doWriteOnHash(h, buf)
+							id, err := machineid.ID()
+							if err == nil {
+								doWriteOnHash(h, []byte(id))
+							}
+							doWriteOnHash(h, []byte("Berty Doctor"))
+							buf = make([]byte, 8)
+							binary.LittleEndian.PutUint64(buf, uint64(runtime.NumCPU()))
+							doWriteOnHash(h, buf)
+							doWriteOnHash(h, []byte(runtime.Version()))
+							key = strconv.FormatUint(binary.LittleEndian.Uint64(h.Sum(nil)), 36)
+						}
+
+						// Advertise
+						{
+							duration, err := disc.Advertise(ctx, key, discovery.TTL(timeout))
+							if err != nil {
+								rtr.message = fmt.Sprintf("Error advertising on %q: %s", tgtMaddr, err)
+								return
+							}
+							// We want at least an Hour of time from the RDVP, else our timeout is fine.
+							if duration < timeout || timeout >= time.Hour {
+								rtr.message = fmt.Sprintf("Wrong time (%d) while advertising on %q", duration, tgtMaddr)
+								disc.Unregister(ctx, key) //nolint:errcheck
+								return
+							}
+						}
+
+						// FindPeers
+						hid := host.ID()
+						{
+							// ReturnChannel
+							rc, err := disc.FindPeers(ctx, key)
+							if err != nil {
+								rtr.message = fmt.Sprintf("Error finding peers on %q: %s", tgtMaddr, err)
+								disc.Unregister(ctx, key) //nolint:errcheck
+								return
+							}
+
+							for v := range rc {
+								if hid == v.ID {
+									// Success !
+									goto TryingCheckUnregister
+								}
+							}
+							// Error we weren't able to find us
+							rtr.message = fmt.Sprintf("Wasn't able to find us on maddr %q", tgtMaddr)
+							disc.Unregister(ctx, key) //nolint:errcheck
+							return
+						}
+
+					TryingCheckUnregister:
+						// Unregister
+						{
+							err = disc.Unregister(ctx, key)
+							if err != nil {
+								rtr.message = fmt.Sprintf("Error unregistering us on %q: %s", tgtMaddr, err)
+								return
+							}
+						}
+
+						// FindPeers again but this time we shouldn't be there.
+						{
+							// ReturnChannel
+							rc, err := disc.FindPeers(ctx, key)
+							if err != nil {
+								rtr.message = fmt.Sprintf("Error finding peers after unregister on %q: %s", tgtMaddr, err)
+								return
+							}
+
+							for v := range rc {
+								if hid == v.ID {
+									rtr.message = fmt.Sprintf("Finding us after unregister on %q", tgtMaddr)
+									return
+								}
+							}
+						}
+						rtr.success = true
+					}(lenAddrs)
+				}
+				// Waiting for defers.
+				lwg.Wait()
+			}(lenPis)
+		}
+		wg.Wait()
+	}
+
+	// Checking for results
+	{
+		// Does at least one were successful ?
+		var success bool
+		for _, v := range errs {
+			if v.success {
+				success = true
+				break
+			}
+		}
+		talkLock.Lock()
+		for _, server := range errs {
+			if server.success {
+				fmt.Printf("[+] %s     ", green("OK"))
+			} else {
+				if success {
+					fmt.Printf("[-] %s   ", yellow("WARN"))
+					warnc++
+				} else {
+					fmt.Printf("[-] %s  ", red("ERROR"))
+					errc++
+				}
+			}
```

Comment:
```
[+] OK     RDVP QmdT7AmhhnbuwvCpa5PH1ySK9HJVB82jr3fo1bxMxBPW6p:
  [+] "/ip4/51.159.21.214/tcp/4040"
  [-] "/ip4/51.159.21.214/udp/4040/quic": Error advertising on "/ip4/51.159.21.214/udp/4040/quic": failed to dial QmdT7AmhhnbuwvCpa5PH1ySK9HJVB82jr3fo1bxMxBPW6p: all dials failed
  * [/ip4/51.159.21.214/udp/4040/quic] NO_ERROR: Handshake did not complete in time
[+] OK     RDVP 12D3KooWHhDBv6DJJ4XDWjzEXq6sVNEs6VuxsV1WyBBEhPENHzcZ:
  [+] "/ip4/51.15.25.224/tcp/4040"
  [-] "/ip4/51.15.25.224/udp/4040/quic": Error advertising on "/ip4/51.15.25.224/udp/4040/quic": failed to dial 12D3KooWHhDBv6DJJ4XDWjzEXq6sVNEs6VuxsV1WyBBEhPENHzcZ: all dials failed
  * [/ip4/51.15.25.224/udp/4040/quic] NO_ERROR: Handshake did not complete in time
[-] WARN   RDVP 12D3KooWRpyQpZtUmY5ZktEMgzuhNoWC1C9zokDjLVahNMy3g48u:
  [-] "/ip4/51.75.127.200/udp/4141/quic": Error advertising on "/ip4/51.75.127.200/udp/4141/quic": failed to dial 12D3KooWRpyQpZtUmY5ZktEMgzuhNoWC1C9zokDjLVahNMy3g48u: all dials failed
  * [/ip4/51.75.127.200/udp/4141/quic] CRYPTO_ERROR: peer IDs don't match
```

completely unreadable

feel free to add a `-v` flag to get more details, but by default we just check that a computer can build and run berty

if we only need one rdvp server to be up & running, then:

* 1 line OK if all servers are up & running
* 1 line Warn if some servers are up and some are down
* 1 line ERROR if no server is up & running